internal/server/http/proxy.go

```diff
@@ -0,0 +1,32 @@
+package http
+
+import (
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+
+	killgrave "github.com/friendsofgo/killgrave/internal"
+)
+
+// Proxy represent reverse proxy server.
+type Proxy struct {
+	server *httputil.ReverseProxy
+	mode   killgrave.ProxyMode
+}
+
+// NewProxy creates new proxy server.
+func NewProxy(rawurl string, mode killgrave.ProxyMode) (*Proxy, error) {
+	u, err := url.Parse(rawurl)
+	if err != nil {
+		return nil, err
+	}
+	reverseProxy := httputil.NewSingleHostReverseProxy(u)
+	return &Proxy{reverseProxy, mode}, nil
+}
+
+// Handler returns handler that sends request to another server.
+func (p *Proxy) Handler() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		p.server.ServeHTTP(w, r)
```

Comment:
you need to update the headers to allow for SSL redirection:

```go
                r.URL.Host = p.url.Host
		r.URL.Scheme = p.url.Scheme
		r.Header.Set("X-Forwarded-Host", r.Header.Get("Host"))
		r.Host = p.url.Host
```

For that you will need to persist the result of url.Parse into the proxy struct.

Reply:
I tested with the comment that I suggested and, all working very fine :)